File: Marvellous_Logic_Building_Applications_Data/Java Programms/SinglyLL.java
/*Single Linked List Implementation using java*/

class Node
{
	int data;
	Node link;

	Node(int data, Node link)
	{
		this.data=data;
		this.link=link;
	}
}

class SinglyLL
{
	static Node root=null;
	static java.util.Scanner sc= new java.util.Scanner(System.in);
	public static void main(String[] args)
	{
		
		int ch;
		while(true)
		{
			System.out.println("1. Append\n");
			System.out.println("2. Display\n");
			System.out.println("3. Quit\n");
			System.out.println("Enter Your choice: ");
			ch=sc.nextInt();

			switch(ch)
			{
				case 1: SinglyLL.Append();
						break;
				case 2: SinglyLL.Display();
						break;
				case 3: System.exit(1);

				default: System.out.println("Invalid choice..!!");

			}
		}
	}

	static void Append()
	{
		Node temp;
		int ele;
		System.out.println("Enter Node data: ");
		ele=sc.nextInt();

		temp=new Node(ele, null);

		if(root == null)
		{
			root=temp;
		}
		else
		{
			Node p=root;

			while(p.link !=null){
				p=p.link;
			}
			p.link=temp;
		}

	}

	static void Display()
	{
		Node temp;
		temp=root;

		if(temp==null)
		{
			System.out.println("List is Empty..!!");
		}
		else
		{
			while(temp != null)
			{
				System.out.print("|" + temp.data + "|->\t");
				temp=temp.link;
			}
			System.out.println("Null");
		}
	}


}

File: Marvellous_Logic_Building_Applications_Data/Data_Structure/SinglyLL.c
#include<stdio.h>
#include<stdlib.h>
#include<conio.h>

struct node 
{
	int data;
	struct node* link;
};

typedef struct node NODE;
typedef struct node* PNODE;

struct node* root=NULL;

void Display(PNODE temp)
{
    if(temp==NULL)
	{
		printf("List is Empty..!!\n");
	}
	else
	{
	    while (temp != NULL)
	    {
	        printf("|%d| -> ",temp->data);
	        temp = temp -> link;
	    }
    	printf("NULL\n");
	}
}

int Length(PNODE temp)
{
	int Count=0;
	temp=root;
	
	while(temp != NULL)
	{
		Count++;
		temp= temp->link;
	}
	return Count;
}

void AddAtBegin()
{
	PNODE temp;
	temp=(PNODE)malloc(sizeof(NODE));
	
	printf("Enter Node Data: ");
	scanf("%d", &temp->data);
	
	temp->link=NULL;
	
	if(root==NULL)
	{
		root=temp;
	}
	else
	{
		temp->link=root;
		root=temp;
	}
}

void Append()
{
	PNODE temp=NULL;
	
	temp=(PNODE)malloc(sizeof(NODE));
	
	printf("Enter the element: ");
	scanf("%d", &temp->data);
	
	temp->link=NULL;
	
	if(root == NULL)
	{
		root= temp;
	}
	else
	{
		PNODE p;
		p=root;
		
		while(p->link!=NULL)
		{
			p= p->link;
		}
		p->link= temp;
	}
}

int main()
{
	struct node data;
	int iCnt=0;
	Append();
	Append();
	Append();
	printf("Elemenet of Linked list are\n");
    Display(root);
    
    AddAtBegin();
    
    Display(root);
    iCnt=Length(root);
    printf("Lenght of list is: %d\n", iCnt);
	return 0;
}
